pandas/tools/merge.py

```diff
@@ -1626,7 +1626,11 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
                 clean_objs.append(v)
             objs = clean_objs
             name = getattr(keys, 'name', None)
-            keys = Index(clean_keys, name=name)
+            # GH 14848
+            # Don't pass name when creating index (# GH 14252)
+            # So that if keys are tuples, name isn't checked
```

Comment:
i c
well that definitely is a bug; a string is NOT a list_like so it SHOULD raise on your example (and not try treat the string as a list)

t